tokio-macros/src/entry.rs

```diff
@@ -393,9 +393,38 @@ fn parse_knobs(mut input: syn::ItemFn, is_test: bool, config: FinalConfig) -> To
                 .block_on(body);
         }
     };
+
+    // For test functions wrap the body in a `Box` to reduce the amount
+    // `Runtime::block_on` (and related functions) copies we generate during
+    // compilation due to the generic parameter `F` (the future to block on).
+    // This could have an impact on performance, but because it's only for
+    // testing and the runtime already boxes the future to run it, it's unlikely
+    // to be very large.
+    //
+    // We don't do this for the main function as it should only be used once so
+    // there will be no benefit.
+    let body = if is_test {
+        let output_type = match &input.sig.output {
+            // For functions with no return value syn doesn't print anything,
+            // but that doesn't work as `Output` for our boxed `Future`, so
+            // default to `()` (the same type as the function output).
+            syn::ReturnType::Default => quote! { () },
+            syn::ReturnType::Type(_, ret_type) => quote! { #ret_type },
+        };
+        quote! {
+            let body = async #body;
+            ::tokio::pin!(body);
+            let body: ::std::pin::Pin<&mut dyn ::std::future::Future<Output = #output_type>> = body;
```

Comment:
This is a breaking change because impl trait is not supported in binding.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ad52f717075b72e126f85dd06b7d0fe2

Reply:
Oh, actually, impl trait doesn't seem to be allowed in tokio::test, so I think this is fine as is.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b242bc072f91f722e3b5d72cfccb5875